File: 04_digital_root.rb
def digital_root(n)
  result = n
  while result > 9
    result = result.digits.sum
  end
  result
end

puts digital_root(0)          # 0
puts digital_root(16)         # 7
puts digital_root(942)        # 6
puts digital_root(132189)     # 6
puts digital_root(493193)     # 2

File: 04_digital_root.js
function digitalRoot(n) {
  return n == 0 ? 0 : 1 + ((n - 1) % 9);
}

console.log(digitalRoot(0)); // 0
console.log(digitalRoot(16)); // 7
console.log(digitalRoot(942)); // 6
console.log(digitalRoot(132189)); //6
console.log(digitalRoot(493193)); //2
console.log(digitalRoot(19930221)); //9
